ystia/yorc
# Feature/gh 282 vault bootstrap

# Pull Request description

## Description of the change

Installing and configuring a vault to store infrastructure credentials, when flag `--insecure` is not used.

### What I did

Used updated components from the Ystia Forge and Alien4Cloud CSAR public library, bundled here in `commands/bootstrap/resources/topology/tosca_types.zip`
to:
* install/configure Hashicorp vault,
* update Yorc configuration to connect to this Vault, and reference secrets stored in this Vault instead of having Infrastructure credentials stored in plain text in Yorc configuration.

For the Yorc HA deployment bootstrap, as the CSAR public library component doesn't yet implement a HA vault, the bootstrap will deploy a single vault instance.

When bootstrap flag `--insecure` is specified, no vault will be installed.

### How to verify it

Use the bootstrap command to install a setup on OpenStack, AWS or Google Cloud.
When the install is done, check Yorc configuration file `/etc/yorc/config.yorc.yaml` to verify that the following values are not stored in plain text, but are references to a secret :
  * AWS access key and secret key on AWS,
  * user name, tenant name and password on OpenStack
  * credentials (service account private keys) on Google Cloud

### Description for the changelog

Bootstrap Yorc with a Vault instance ([GH-282](https://github.com/ystia/yorc/issues/282))

## Applicable Issues

https://github.com/ystia/yorc/issues/282
